packages/rich-text/src/component/index.native.js

```diff
@@ -977,6 +973,16 @@ RichText.defaultProps = {
 	tagName: 'div',
 };
 
+const withFormatTypes = ( WrappedComponent ) => ( props ) => {
+	const { formatTypes } = useFormatTypes( {
```

Comment:
This hook returns more useful things than just the "allowed formats". So it might be good to consider using these in mobile to consolidate that said, I'm not sure what these things are about, I'll leave it to RichText experts. For now, I'm just making sure the logic to filter the available format types is applied in mobile as well.

Reply:
Why the HoC?

Reply:
because the mobile component is not a functional component yet.

Reply:
Hm, any way to do it for mobile only? This will block the rich text as a hook effort, although it's not a priority.

Reply:
This is just for mobile

Reply:
Oh, sorry, I missed the file name...